File: internal/brew/getAppInfo.js
import execBrew from "./execBrew";

// brew info --json=v2 libreoffice

export default async function getAppInfo(appName) {
  const raw = await execBrew(`cask info ${appName}`);

  let nameNext = false;
  let artifactsNext = false;
  const info = raw.reduce(
    (obj, line) => {
      if (line === "==> Name") {
        nameNext = true;
      } else if (nameNext) {
        nameNext = false;
        return {
          ...obj,
          name: line,
        };
      } else if (line === "==> Artifacts") {
        artifactsNext = true;
      } else if (artifactsNext) {
        return {
          ...obj,
          artifacts: [...obj.artifacts, line],
        };
      }

      return obj;
    },
    {
      name: "",
      artifacts: [],
    }
  );

  return info;
}

File: internal/brew/GetAppInfo.go
package brew

import (
	"encoding/json"
)

type Artifact struct {
	App []string
	Pkg []string
}

type AppInfo struct {
	Name      []string
	Artifacts []Artifact
}

type appInfoItems struct {
	Casks []AppInfo
}

func GetAppInfo(name string) (*AppInfo, error) {
	infoJSON, err := ExecBrew("info", "--json=v2", name)
	if err != nil {
		return nil, err
	}

	var appInfo appInfoItems
	err = json.Unmarshal([]byte(infoJSON), &appInfo)

	if err != nil {
		return nil, err
	}

	if len(appInfo.Casks) == 0 {
		return &AppInfo{}, nil
	}

	return &appInfo.Casks[0], nil
}
